Sources/TensorFlow/Core/TensorShape.swift

```diff
@@ -110,6 +110,20 @@ public extension TensorShape {
         get { return TensorShape(dimensions[bounds]) }
         set { dimensions[bounds] = ArraySlice(newValue.dimensions) }
     }
+
+    /// Returns the reduced shape along the given axes. For each axis, the dimension is set to 1.
+    ///
+    /// Adapted from `reduced_shape` in Python TensorFlow:
+    /// https://github.com/tensorflow/tensorflow/blob/8023e0736756b93308cbe847e5e6afbfb625f8d8/tensorflow/python/ops/math_ops.py#L3490
+    func reducedShape(axes: [Int32]) -> TensorShape {
```

Comment:
Actually, since the logic of `reducedShape` is so simple, how about inlining its defining in `_vjpProduct`?
Sorry for the thrash. Adding `reducedShape` as an extension method on `TensorShape` is rather invasive.

Reply:
Yeah, no problem.